src/app/utils/graph-toolkit-lookup.ts

```diff
@@ -1,7 +1,33 @@
+import { SeverityLevel } from '@microsoft/applicationinsights-web';
 import templates from '../../graph-toolkit-examples';
+import { telemetry } from '../../telemetry';
+import { LINK_ERROR } from '../../telemetry/error-types';
 import { IQuery } from '../../types/query-runner';
+import { sanitizeQueryUrl } from './query-url-sanitization';
 import { parseSampleUrl } from './sample-url-generation';
 
+
+export async function validateToolkitUrl(url: string, sampleQuery: IQuery): Promise<void> {
+  await fetch(url)
+      .then(response => {
+        if (!response.ok) {
+          throw Error(response.statusText);
+        }
+      })
+      .catch(error => {
+        const sanitizedUrl = sanitizeQueryUrl(sampleQuery.sampleUrl);
+        telemetry.trackException(
+          new Error(LINK_ERROR),
+          SeverityLevel.Error,
+          {
+            ComponentName: 'Graph toolkit link',
```

Comment:
What do you think of making the component names constants? It will make it easier to update them and will give us the added benefit of knowing what components are instrumented.

Reply:
I've done this, thanks